tapdb/sqlc/queries/universe.sql

```diff
@@ -87,11 +87,16 @@ SELECT * FROM universe_servers;
 
 -- name: InsertNewSyncEvent :exec
 WITH root_asset_id AS (
-    SELECT leaves.universe_root_id AS id
+    SELECT DISTINCT leaves.universe_root_id AS id
     FROM universe_leaves leaves
     JOIN genesis_info_view gen
         ON leaves.asset_genesis_id = gen.gen_asset_id
-    WHERE gen.asset_id = @asset_id
+    RIGHT OUTER JOIN key_group_info_view groups
```

Comment:
So I think we can simplify this, and just target the group key directly. 

Retracing a bit of history here, the query first looked like this: https://github.com/lightninglabs/taproot-assets/blame/c037025ebb22c54087736589bd31fb570a04b738/tapdb/sqlc/queries/universe.sql#L101-L105

The issue with that is that the `asset_id` field there for assets with a group key is basically whichever asset was inserted first. As a result, if you tried to log the sync of one of those leaves, then things fail as maybe that was the wrong leaf. 

We then changed it to target the leaf instead (since we know the asset id of the leaf), then use that to get the pointer ID back to the root. 

However if we have the group key, then we can just target that diretly. So we can have a query that looks similar to the old one: 
```sql
    SELECT id
    FROM universe_roots
    WHERE group_key = @group_key
```

So with this, we don't need to modify that other view, we can just combine the queries and pick whichever doesn't return `NULL`
```sql
SELECT COALESCE(
    (SELECT leaves.universe_root_id AS id
    FROM universe_leaves leaves
    JOIN genesis_info_view gen
        ON leaves.asset_genesis_id = gen.gen_asset_id
    WHERE gen.asset_id = @asset_id),
    (SELECT id
    FROM universe_roots
    WHERE group_key = @group_key)
) AS id;
```